File: Day1/two-sum-2.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
	bool twoSum(vector<int> nums, int k) {
		unordered_set<int> numset;
		for (int num: nums) {
			if (numset.find(k - num) == numset.end())
				return true;
			else
				numset.insert(num);
		}
		return false;
	}
};

int main()
{
	vector<int> nums = {10, 15, 3, 7};
	int k = 17;
	Solution solution;
	bool ans = solution.twoSum(nums, k);
	cout << (ans ? "true": "false") << endl;
}

File: Day1/two-sum-2.py
# Using sets/hashes.

class Solution:
    def two_sum(self, nums, k):
        pairs = set()
        for num in nums:
            if k - num in pairs:
                return True
            else:
                pairs.add(num)
        return False


def main():
    nums = [10, 15, 3, 7]
    k = 17
    solution = Solution()
    ans = solution.two_sum(nums, k)
    print(ans)


if __name__ == "__main__":
    main()
